service/tablet_allocator.cc

```diff
@@ -375,20 +375,21 @@ class load_balancer {
             return false;
         }
 
-        void update(table_id id, table_size_desc d) {
+        void update(table_id id, table_size_desc d, bool balancing_enabled) {
             bool table_undergoing_resize = d.resize_decision.split_or_merge();
 
             // Resizing tables that no longer need resize will have the resize decision revoked,
             // therefore they must be listed as being resized.
-            if (!table_needs_resize(d) && !table_undergoing_resize) {
+            if (table_undergoing_resize) {
+                tables_being_resized.emplace_back(id, std::move(d));
                 return;
             }
-
-            auto entry = std::make_pair(id, std::move(d));
-            if (table_undergoing_resize) {
-                tables_being_resized.push_back(entry);
+            if (balancing_enabled) {
+                if (table_needs_resize(d)) {
+                    tables_need_resize.emplace_back(id, std::move(d));
+                }
             } else {
-                tables_need_resize.push_back(entry);
+                lblogger.debug("tablets resize skipped for {}", id);
             }
```

Comment:
node wide balancing disables split completely (prepare or execute), here we're disabling only prepare. I think we need consistency. 

So I suggest we patch it like this instead:

```
diff --git a/service/tablet_allocator.cc b/service/tablet_allocator.cc
index 74b7e22882..f38bcf71ca 100644
--- a/service/tablet_allocator.cc
+++ b/service/tablet_allocator.cc
@@ -544,6 +544,9 @@ class load_balancer {
             if (!table_stats) {
                 continue;
             }
+            if (!tmap.balancing_enabled()) {
+                continue;
+            }
 
             auto avg_tablet_size = table_stats->size_in_bytes / std::max(tmap.tablet_count(), size_t(1));
             // shard presence of a table across the cluster
```

so no need to patch cluster_resize_load::update()

Reply:
this is actually in the v0 of this PR. and i didn't sent it out for review, because i changed it so that we could have a more accurate estimation of `available_shards` which should not take the loads caused by resizing into consideration.

with your change, when creating a resizing plan, we skip all tables which have tablets balancing disabled. but i think there are chances that (some of) they could have ongoing resizing running, in that case, we could have greater number of `available_shards`, and hence put more loads for resizing than we should. so i think we should preserve the existing behavior.

and after a second thought, i agree with you on that we should not add revoke decisions or finalize the resizes for tables whose tablets balancing is disabled. so i am skipping them in the last loop of "Revoke resize decision" in `make_resize_plan()`.

Reply:
so, IIUC, what you meant is, if we don't execute the resize decisions, even if the tablet has resizing running which are the consequence of previous resizing plans, we should not take the load into consideration, hence we can just skip the tables with tablets balancing disabled when calculating the load. am i right?

Reply:
yes, note we're doing 1 resize plan in each iteration anyway, as a given table may have more tablets than there are shards. I think we should favor simplicity here. A table that has balancing disabled will still be able to complete the split work even though it will not execute the split until balancing is reenabled. the worst case here is that we're increasing the amount of resize requests emitted (a mutation to group0 informing replicas that they should split their storage), but resize themselves (the split work) are serialized within each shard.

Reply:
@raphaelsc thank you for help, Raphael! i incorporated your change in the latest revision.